glyphs-reader/src/font.rs

```diff
@@ -66,7 +60,114 @@ pub struct Font {
     pub date: Option<String>,
 
     // master id => { (name or class, name or class) => adjustment }
-    pub kerning_ltr: BTreeMap<String, BTreeMap<(String, String), i32>>,
+    pub kerning_ltr: Kerning,
+}
+
+/// master id => { (name or class, name or class) => adjustment }
+#[derive(Clone, Debug, Default, PartialEq, Hash)]
+pub struct Kerning(BTreeMap<String, BTreeMap<(String, String), i32>>);
+
+impl Kerning {
+    pub fn get(&self, master_id: &str) -> Option<&BTreeMap<(String, String), i32>> {
+        self.0.get(master_id)
+    }
+
+    pub fn keys(&self) -> impl Iterator<Item = &String> {
+        self.0.keys()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (&String, &BTreeMap<(String, String), i32>)> {
+        self.0.iter()
+    }
+
+    fn insert(
+        &mut self,
+        master_id: String,
+        lhs_class_or_group: String,
+        rhs_class_or_group: String,
+        kern: i64,
+    ) {
+        *self
+            .0
+            .entry(master_id)
+            .or_default()
+            .entry((lhs_class_or_group, rhs_class_or_group))
+            .or_default() = kern as i32;
+    }
+}
+
+/// Hand-parse because it's a bit weird
```

Comment:
curious, what is weird about kerning dictionary that requires it to be hand-parsed?

Reply:
It's not a structure with named fields and consistent form, e.g. kern.left_side = blah, we use blah as the field name.